### PR TITLE
WF-64807-Adding UG documentation to IncludeNewLineCharacters API

### PR DESCRIPTION
**Pull request by chrispojunior**
<table style="width: 521.306px;">
<tbody>
<tr>
<td style="width: 156px;">Task Link</td>
<td style="width: 349.306px;">https://syncfusion.atlassian.net/browse/WF-64807</td>
</tr>
<tr>
<td style="width: 156px;">Description</td>
<td style="width: 349.306px;">Add UG document to&nbsp; customize Table of contents</td>
</tr>
<tr>
<td style="width: 156px;">Solution</td>
<td style="width: 349.306px;">Added code snippet to customize table of contents</td>
</tr>
<tr>
<td style="width: 156px;">Content review confirmation</td>
<td style="width: 349.306px;">Not yet</td>
</tr>
</tbody>
</table>

### CODE REVIEW

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 1]
```diff
@@ -581,10 +581,358 @@ document.Close();
...
+
+{% highlight c# tabtitle="C#" %}
+//Create a new Word document.
+using (WordDocument document = new WordDocument())
+{
+    //Add a section into the Word document.
+    IWSection section = document.AddSection();
+    //Add a paragraph into the created section.
+    IWParagraph paragraph = section.AddParagraph();
+    // Append table of content to the end of the paragraph with the specified lower and upper heading levels.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:604]: //Appends the TOC field with LowerHeadingLevel and UpperHeadingLevel to determines the TOC entries

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 2]
```diff
@@ -581,10 +581,358 @@ document.Close();
...
+    TableOfContent tableOfContent = paragraph.AppendTOC(2, 5);
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:605]: paragraph.AppendTOC(1, 3);

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 3]
```diff
@@ -581,10 +581,358 @@ document.Close();
...
+    //Set starting heading level of the table of contents.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:606]: //Set lower heading level for TOC

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 4]
```diff
@@ -581,10 +581,358 @@ document.Close();
...
+    tableOfContent.LowerHeadingLevel = 2;
+    //Set ending heading level of the table of contents.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:608]: //Set upper heading level for TOC.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 5]
```diff
@@ -581,10 +581,358 @@ document.Close();
...
+    tableOfContent.UpperHeadingLevel = 5;
+    //Use default heading styles
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:610]: //Enable a flag to use default heading styles in TOC entries.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 6]
```diff
@@ -581,10 +581,358 @@ document.Close();
...
+    tableOfContent.UseHeadingStyles = true;
+    //Show page numbers in table of content.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:612]: //Enable a flag to show page numbers in TOC entries.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 7]
```diff
@@ -581,10 +581,358 @@ document.Close();
...
+    tableOfContent.IncludePageNumbers = true;
+    //Set page numbers to right alignment.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:614]: //Disable a flag to align page numbers after to the TOC entries.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 8]
```diff
@@ -581,10 +581,358 @@ document.Close();
...
+    tableOfContent.RightAlignPageNumbers = false;
+    //Set hyperlinks for the TOC levels.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:616]: //Disable a flag to preserve TOC entries without hyperlinks

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 9]
```diff
@@ -581,10 +581,358 @@ document.Close();
...
+    tableOfContent.UseHyperlinks = false;
+    //Add a paragraph into the section.
+    paragraph = section.AddParagraph();
+    //Append text.
+    paragraph.AppendText("First ");
+    //Append line break.
+    paragraph.AppendBreak(BreakType.LineBreak);
+    paragraph.AppendText("Chapter");
+    //Include new line to preserve newline characters TableOfContent.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:625]: //Enable a flag to include new line characters in TOC entries.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 10]
```diff
@@ -581,10 +581,358 @@ document.Close();
...
+    //Add the text into the new paragraph of the section.
+    section.AddParagraph().AppendText("AdventureWorks Cycles, the fictitious company on which the AdventureWorks sample databases are based, is a large, multinational manufacturing company.");
+    //Add a paragraph to the section.
+    paragraph = section.AddParagraph();
+    //Set outline level for paragraph.
+    paragraph.ParagraphFormat.OutlineLevel = OutlineLevel.Level2;
+    //Append text.
+    paragraph.AppendText("Outline Level Paragraph");
+    //Set the outline levels.
+    tableOfContent.UseOutlineLevels = true;
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:660]: //Enable a flag to consider outline level paragraphs in TOC entries.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 11]
```diff
@@ -581,10 +581,358 @@ document.Close();
...
+    //Add a section into the Word document.
+    section = document.AddSection();
+    //Add a paragraph to the section.
+    paragraph = section.AddParagraph();
+    //Append a field to the paragraph.
+    paragraph.AppendField("Table of Entry Field", FieldType.FieldTOCEntry);
+    //Indicate whether to use table entry fields.
+    tableOfContent.UseTableEntryFields = true;
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:668]: //Enable a flag to use table entry fields in TOC entries

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 12]
```diff
@@ -581,10 +581,358 @@ document.Close();
...
 {% endhighlight %}
 
-{% endtabs %}  
+{% endtabs %}
 
 You can download a complete working sample from [GitHub](https://github.com/SyncfusionExamples/DocIO-Examples/tree/main/Table-Of-Contents/TOC-with-user-defined-styles).
 
+## Applying switches for table of contents
+
+Switches are used to change the default settings in word document. TOC will determine the TOC entries based on the TOC switches. You can set heading levels, specify whether to use default heading styles, show page numbers, set page numbers to right alignment, set hyperlinks for TOC levels, preserve line break and text wrapping break in TOC, set outline levels for paragraph and specify whether to use table of entry fields.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:590]: You can modify the preservation of TOC entries in Word document by applying switches to table of contents. You can apply switches to preserve TOC entries as hyperlink or not, use default heading styles, preserve page numbers or not, aligning the page numbers, preserve new line characters, preserve outline level paragraphs, and preserve table of entry fields in the TOC entries using Word (DocIO) library.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 13]
```diff
@@ -581,10 +581,358 @@ document.Close();
 //https://help.syncfusion.com/file-formats/docio/create-word-document-in-xamarin#helper-files-for-xamarin
 {% endhighlight %}
 
-{% endtabs %}  
+{% endtabs %}
 
 You can download a complete working sample from [GitHub](https://github.com/SyncfusionExamples/DocIO-Examples/tree/main/Table-Of-Contents/TOC-with-user-defined-styles).
 
+## Applying switches for table of contents
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:588]: Apply switches for table of contents

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 14]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
 {% endhighlight %}
 
-{% endtabs %}  
+{% endtabs %}
 
 You can download a complete working sample from [GitHub](https://github.com/SyncfusionExamples/DocIO-Examples/tree/main/Table-Of-Contents/TOC-with-user-defined-styles).
 
+## Apply switches for table of contents
+
+You can modify the preservation of TOC entries in Word document by applying switches to table of contents. You can apply switches to preserve TOC entries as hyperlink or not, use default heading styles, preserve page numbers or not, aligning the page numbers, preserve new line characters, preserve outline level paragraphs, and preserve table of entry fields in the TOC entries using Word (DocIO) library.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:590]: You can modify the preservation of TOC entries in a Word document by applying switches to the table of contents, and you can also apply switches to preserve TOC entries as hyperlinks or not, use the default heading styles, preserve page numbers or not, align the page numbers, preserve new line characters, preserve outline level paragraphs, and preserve table of entry fields in the TOC entries using Word (DocIO) library.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 15]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+
+The following code example illustrates how to apply switches for table of contents.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:592]: The following code example illustrates how to apply switches for the table of contents.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 16]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+
+{% tabs %} 
+
+{% highlight c# tabtitle="C#" %}
+//Create a new Word document.
+using (WordDocument document = new WordDocument())
+{
+//Add a section into the Word document.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:600]: Add a section to the Word document.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 17]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+IWSection section = document.AddSection();
+//Add a paragraph into the created section.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:602]: Add a paragraph to the created section.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 18]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+IWParagraph paragraph = section.AddParagraph();
+//Append the TOC field with LowerHeadingLevel and UpperHeadingLevel to determines the TOC entries.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:604]: Append the TOC field with LowerHeadingLevel and UpperHeadingLevel to determine the TOC entries.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 19]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+TableOfContent tableOfContent = paragraph.AppendTOC(1, 3);
+//Set lower heading level for TOC.
+tableOfContent.LowerHeadingLevel = 2;
+//Set upper heading level for TOC.
+tableOfContent.UpperHeadingLevel = 5;
+//Enable a flag to use default heading styles in TOC entries.
+tableOfContent.UseHeadingStyles = true;
+//Enable a flag to show page numbers in TOC entries.
+tableOfContent.IncludePageNumbers = true;
+//Disable a flag to align page numbers after to the TOC entries.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:614]: Disable a flag to align page numbers after the TOC entries.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 20]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+tableOfContent.RightAlignPageNumbers = false;
+//Disable a flag to preserve TOC entries without hyperlinks.
+tableOfContent.UseHyperlinks = false;
+//Add a paragraph into the section.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:618]: Add a paragraph to the section.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 21]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+//Append text.
+paragraph.AppendText("First ");
+//Append line break.
+paragraph.AppendBreak(BreakType.LineBreak);
+paragraph.AppendText("Chapter");
+//Enable a flag to include new line characters in TOC entries.
+tableOfContent.IncludeNewLineCharacters = true;
+//Set a built-in heading style.
+paragraph.ApplyStyle(BuiltinStyle.Heading2);
+//Add a section into the Word document.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:629]: Add a section to the Word document.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 22]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+section = document.AddSection();
+//Add a paragraph to the section.
+paragraph = section.AddParagraph();
+//Append text.
+paragraph.AppendText("Second ");
+paragraph.AppendText("Chapter");
+//Sets a built-in heading style.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:636]: Set a built-in heading style.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 23]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+paragraph.ApplyStyle(BuiltinStyle.Heading1);
+//Add a section into the Word document.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:638]: Add a section to the Word document.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 24]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+paragraph.AppendText("Chapter");
+//Set a built-in heading style.
+paragraph.ApplyStyle(BuiltinStyle.Heading2);
+section = document.AddSection();
+paragraph = section.AddParagraph();
+paragraph.AppendText("Fourth ");
+paragraph.AppendText("Chapter");
+//Set a built-in heading style.
+paragraph.ApplyStyle(BuiltinStyle.Heading3);
+//Add the text into the new paragraph of the section.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:651]: Add the text to the new paragraph of the section.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 25]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+section.AddParagraph().AppendText("AdventureWorks Cycles, the fictitious company on which the AdventureWorks sample databases are based, is a large, multinational manufacturing company.");
+//Add a paragraph to the section.
+paragraph = section.AddParagraph();
+//Set outline level for paragraph.
+paragraph.ParagraphFormat.OutlineLevel = OutlineLevel.Level2;
+//Append text.
+paragraph.AppendText("Outline Level Paragraph");
+//Enable a flag to consider outline level paragraphs in TOC entries.
+tableOfContent.UseOutlineLevels = true;
+//Add a section into the Word document.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:661]: Add a section to the Word document

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 26]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+section = document.AddSection();
+//Add a paragraph to the section.
+paragraph = section.AddParagraph();
+//Append a field to the paragraph.
+paragraph.AppendField("Table of Entry Field", FieldType.FieldTOCEntry);
+//Enable a flag to use table entry fields in TOC entries.
+tableOfContent.UseTableEntryFields = true;
+//Update the table of content.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:669]: Update the table of contents.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 27]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+tableOfContent.UseTableEntryFields = True
+'Update the table of content.
+document.UpdateTableOfContents()
+'Save a Word document.
+document.Save("Sample.docx")
+End Using
+{% endhighlight %}
+
+{% highlight c# tabtitle="UWP" %}
+//DocIO supports Table of contents in WPF, Windows Forms platforms alone
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:756]: Keep full stop at the end

**Comment by laksmipriyamurugesan** [comment]
Reviewed the PR.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 28]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+//Update the table of content.
+document.UpdateTableOfContents();
+//Save a Word document to MemoryStream.
+MemoryStream outputStream = new MemoryStream();
+document.Save(outputStream, FormatType.Markdown);
+//Save a stream as a file in the device and invoke it for viewing.
+Xamarin.Forms.DependencyService.Get<ISave>().SaveAndView("Sample.docx", "application/msword", outputStream);
+}
+
+//Please download the helper files from the below link to save the stream as file and open the file for viewing in Xamarin platform
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:928]: Please download the helper files from the following link to save the stream as a file and open the file for viewing on Xamarin platform.

Keep full stop at the end.

File: File-Formats/DocIO/Working-with-Table-Of-Contents.md [hunk 29]
```diff
@@ -581,10 +581,356 @@ document.Close();
...
+paragraph.ApplyStyle(BuiltinStyle.Heading3);
+//Add the text into the new paragraph of the section.
+section.AddParagraph().AppendText("AdventureWorks Cycles, the fictitious company on which the AdventureWorks sample databases are based, is a large, multinational manufacturing company.");
+//Add a paragraph to the section.
+paragraph = section.AddParagraph();
+//Set outline level for paragraph.
+paragraph.ParagraphFormat.OutlineLevel = OutlineLevel.Level2;
+//Append text.
+paragraph.AppendText("Outline Level Paragraph");
+//Enable a flag to consider outline level paragraphs in TOC entries.
```
Comment (laksmipriyamurugesan) [inline File-Formats/DocIO/Working-with-Table-Of-Contents.md:909]: Enable a flag to consider outline-level paragraphs in TOC entries.

**Review by MohanaselvamJothi — APPROVED** [review]

**Pull request merged**